### PR TITLE
fix(rollout): apply torch compile mode correctly

### LINKED ISSUE #4114: lerobot-rollout --use_torch_compile silently fails: torch.compile called with both mode and options
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

```Shell
LeRobot 0.6.0, NVidia Jetson Orin NX 16, Jetson 7.2
```

### Description

Bug: In lerobot/rollout/context.py (v0.6.0), the compile block passes both mode=cfg.torch_compile_mode and options={"triton.cudagraphs": False} to torch.compile(). PyTorch forbids this combination, so every --use_torch_compile=true run logs Failed to apply torch.compile: Either mode or options can be specified, but both can't be specified at the same time and silently continues in eager mode. The RTC engine still reports "torch.compile enabled" and performs its warmup inferences, masking the failure.

Fix: Drop the options line (or the mode line). Note the current options value would also disable CUDA graphs — the main benefit of reduce-overhead on CPU-bound edge devices.
Impact (measured): SmolVLA fine-tune, Jetson Orin NX 16 GB (JetPack 7.2, torch 2.11.0+cu130, bf16), batch 1: eager 792 ms → compiled(default) 264 ms → compiled(reduce-overhead) 180 ms median. Numerical parity eager vs. compiled: cosine 0.99998.

Repro: any lerobot-rollout ... --use_torch_compile=true; observe the warning in the log.

### Context & Reproduction

_No response_

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

### Additional Info / Workarounds

_No response_

### PR DESCRIPTION
**Pull request by patrickswedish**
## Summary / Motivation

`lerobot-rollout --use_torch_compile=true` currently passes both `mode` and `options` to `torch.compile`, although PyTorch treats those arguments as mutually exclusive. The compile call fails and rollout silently continues in eager mode while RTC still receives the requested compile flag.

This change applies the configured backend and mode without the conflicting options, and tracks whether compilation was actually applied so downstream warm-up behavior reflects reality.

## Related issues

- Fixes #4114

## What changed

- compile `predict_action_chunk` with the configured backend and mode only
- return an explicit success state from the compilation boundary
- disable compile-dependent RTC warm-up behavior when compilation is unavailable or raises
- cover the successful compiler call and failure reporting with regression tests

No public API changes.

## How was this tested

- `uv run pytest -q tests/test_rollout.py` — 26 passed
- `uv run pre-commit run --files src/lerobot/rollout/context.py tests/test_rollout.py` — all applicable hooks passed, including Ruff, typos, secret detection, Bandit, and mypy
- `uv run ruff format --check src/lerobot/rollout/context.py tests/test_rollout.py`
- `git diff --check`

## Checklist

- [x] Linting and formatting run
- [x] Relevant tests pass locally
- [x] Documentation not required; behavior and CLI surface are unchanged
- [ ] CI is green
- [x] Community review completed: #4112

## Reviewer notes

Please focus on the effective compile-state propagation into RTC warm-up. AI assistance was used to draft the implementation and tests; the patch was then validated with the repository's targeted tests and full file-level pre-commit gate.

### CODE REVIEW

**Review by patrickswedish — COMMENTED** [review]
I reviewed the implementation and regression tests. The change removes the conflicting `torch.compile` arguments, preserves the configured compilation mode, and ensures RTC warm-up only treats compilation as active when it was successfully applied. The success and failure paths are both covered by tests. I understand and approve these changes for maintainer review.

**Comment by imstevenpmwork** [comment]
Superseded by https://github.com/huggingface/lerobot/pull/4268
